tests/kafkatest/tests/streams/streams_upgrade_test.py

```diff
@@ -515,11 +515,11 @@ def do_rolling_bounce(self, processor, counter, current_generation):
                                               err_msg="Could not detect 'version probing' attempt at leader " + str(self.leader.node.account))
 
                     if len(self.old_processors) > 0:
-                        log_monitor.wait_until("Sent a version 4 subscription and got version 3 assignment back (successful version probing). Downgrading subscription metadata to received version and trigger new rebalance.",
```

Comment:
Hmm.. I'm not sure if this is the right fix, but maybe upgrading the test when we bump up version is also out side the scope of this PR itself. I'll let @mjsax to take a look and decide how can we fix forward the upgrade-test.

Reply:
If we bump the version, we need to update this to 5 and 4 as already done.

We should also make this more generic and test upgrades from 3 -> 4, 3 -> 5 and 4 -> 5. The current code does only go from latest version to future version. However, generalizing the test should be out of scope of this PR and just changing the expected numbers should be fine for this PR.